### PR TITLE
Lattice resources

### PR DESCRIPTION
**Pull request by aherbst-broad**
## Why

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1709)
The Lattice project will require a staging bucket with appropriate permissions in which they will deposit files for ingest.

## This PR
* Attempts to extract a common "staging bucket" module that can be reused going forward should further consortia require a staging bucket in our infra. 
* Instantiates the Lattice infra using the above module

### CODE REVIEW

File: environments/hca/terraform/buckets.tf [hunk 1]
```diff
@@ -145,5 +145,8 @@ resource google_service_account_iam_binding dataflow_runner_user_binding {
 
   service_account_id = module.hca_dataflow_account.id
   role               = "roles/iam.serviceAccountUser"
-  members            = ["serviceAccount:${module.hca_argo_runner_account.email}"]
+  members            = [
+    "serviceAccount:${module.hca_argo_runner_account.email}",
+    "serviceAccount:${module.hca_dagster_runner_account.email}"
```
Comment (aherbst-broad) [inline environments/hca/terraform/buckets.tf:150]: `terraform plan` showed an inconsistency between what's deployed and our code.

**Review by raaidbroad — APPROVED** [review]
Good idea to put this in a template 👍🏾

**Review by quazi-broad — APPROVED** [review]
lgtm

**Review by ThePletch — APPROVED** [review]
looks good. it looks like we're only granting read access to the dataflow user, though, and it seems like we'd want that user to have write access to our staging buckets as well. am i misinterpreting something there?

**Comment by aherbst-broad** [comment]
@ThePletch good question; "staging areas" are read only in our world (i.e., an external group is handing off data to us). Dataflow should never need to write to that bucket; it'll deposit any transformation results in a different "internal" bucket.

**Pull request merged**